### DISCUSSION TITLE
How to run postdeploy task which persists?

### DISCUSSION THREAD

**Discussion by wojtekkaminski** [dokku/dokku#5246] [Q&A] [answered]
Hi,

I have two apps on dokku:
App A. PHP + Laravel 
App B. Javascript + VueJS 

I need to run some tasks after deploy. It doesn't work. There is no error (I can see my tasks running during deployment) but there is also no effect.

I used app.json, I used Procfile. It still doesn't work.

My tasks:
- App A: php artisan migrate --force && php artisan route:clear && php artisan config:clear
- App B: ln -s ../../config/app.config.js dist/assets/js/app.config.js

In the second example task directory `config` is a permanent storage linked from the host.

Both commands work just fine if I enter the container and run it manually.

Please, please help

**Comment by josegonzalez** [comment, marked as answer]
What do you mean by "it still doesn't work"?

One thing to note, at least for that `ln -s` command (and probably the artisan route/config clear commands), postdeploy tasks happen in a separate container (not any of the existing ones). They wouldn't have any impact on the running containers.

If you want something to happen in a running container and have any of the changes "persist", you'd want to use a `predeploy` task.

Since all apps start in new containers, the artisan route/config clear commands _probably_ don't need to run unless they clear something in an external cache (like a database or something).

The `php artisan migrate` can either go to a `predeploy` command or go to your `Procfile` under the `release` entry. We don't support ampersands in the latter, so my suggestion is predeploy.

**Reply by wojtekkaminski** [reply]
Muchísimas gracias @josegonzalez!! <3 You saved my life!

Until now I didn't know that `postdeploy` and `release` tasks were both runing in different containers. And also, I wrongly though about `predeploy` as a `prebuild`.

Thanks a million!!
